### PR TITLE
Add emitDeclarationOnly. Generate declaration

### PR DESCRIPTION
**Pull request by a-tarasyuk**

### CODE REVIEW

**Comment by a-tarasyuk** [comment]
@DanielRosenwasser could you review this PR? This PR related to the issue https://github.com/Microsoft/TypeScript-Handbook/issues/730

File: README.md [hunk 1]
```diff
@@ -64,7 +64,7 @@ or make sure that you add the appropriate `"devDependencies"` entries to your `p
 Then run
 
 ```sh
-tsc --init --noEmit --allowJs --allowSyntheticDefaultImports --target esnext
+tsc --init --declaration --allowJs --allowSyntheticDefaultImports --target esnext
```
Comment (DanielRosenwasser) [inline README.md]: Why not just set `--declaration` and `--emitDeclarationOnly` here?

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

**Comment by DanielRosenwasser** [comment]
Looks good, just want to see if we can keep the commands cleaner.

File: README.md (hunk 1 above)
Comment (a-tarasyuk) [reply README.md]: @DanielRosenwasser  `emitDeclarationOnly` and `noEmit` cannot be used simultaneously 

> Option 'emitDeclarationOnly' cannot be specified with option 'noEmit'.

that's why I added it directly in npm scripts 

`"type-check": "tsc --noEmit"`
 `"build": "tsc --emitDeclarationOnly ...`

Does that make sense?

Comment (a-tarasyuk) [reply README.md]: @DanielRosenwasser What do you think about my previous comment?

Comment (DanielRosenwasser) [reply README.md]: Agh, okay, then this is fine, but shouldn't the command line initialization specify an `--outDir` or `--declarationDir` instead?

Comment (a-tarasyuk) [reply README.md]: Sure, I'll add `--declarationDir` in README

Comment (a-tarasyuk) [reply README.md]: @DanielRosenwasser Done, could you check, please?

**Review by DanielRosenwasser — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
Thank you so much for the PR! 😃

**Comment by a-tarasyuk** [comment]
@DanielRosenwasser :smiley:. Thank you very much for your support :muscle:

**Pull request merged**